depends: OpenBSD (aarch64) needs gcc (instruction) for libevent

### Current behaviour

The current depends instructions for OpenBSD do not mention `gcc`, which is required to build libevent.

A simple `pkg_add gcc` doesn't cut it either, because that only installs `egcc`: https://unix.stackexchange.com/questions/554752/no-gcc-executable-after-pkg-add-gcc-on-openbsd

### Expected behaviour

Explain what the user needs to do to build depends (or change depends).

A simple symlink seems to do the trick, but I'm not if it's appropriate:

```
ln -s /usr/local/bin/egcc /usr/local/bin/gcc
```

### Steps to reproduce

Fresh OpenBSD installation. Clone repo on master.

```
gmake -C depends NO_QT=1 NO_QR=1 NO_ZMQ=1 NO_USDT=1

gmake: Entering directory '/home/sjors/src/bitcoin/depends'
Configuring libevent...
CMake Deprecation Warning at CMakeLists.txt:22 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


CMake Error at /usr/local/share/cmake/Modules/CMakeDetermineCCompiler.cmake:49 (message):
  Could not find compiler set in environment variable CC:

  gcc.
Call Stack (most recent call first):
  CMakeLists.txt:47 (project)


CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
gmake: *** [funcs.mk:343: /home/sjors/src/bitcoin/depends/aarch64-unknown-openbsd7.7/.libevent_stamp_configured] Error 1
gmake: Leaving directory '/home/sjors/src/bitcoin/depends'
```

### Relevant log output

_No response_

### How did you obtain Bitcoin Core

Compiled from source

### What version of Bitcoin Core are you using?

master

### Operating system and version

OpenBSD 7.7

### Machine specifications

VM inside UTM, host machine is M4 macOS 15.5

--- a/depends/builders/freebsd.mk
+++ b/depends/builders/freebsd.mk
@@ -3,3 +3,7 @@ build_freebsd_CXX=clang++
 
 build_freebsd_SHA256SUM = sha256sum
 build_freebsd_DOWNLOAD = curl --location --fail --connect-timeout $(DOWNLOAD_CONNECT_TIMEOUT) --retry $(DOWNLOAD_RETRIES) -o
+
+# freebsd host on freebsd builder: override freebsd host preferences.
+freebsd_CC = clang
+freebsd_CXX = clang++

--- a/depends/builders/openbsd.mk
+++ b/depends/builders/openbsd.mk
@@ -7,3 +7,7 @@ build_openbsd_DOWNLOAD = curl --location --fail --connect-timeout $(DOWNLOAD_CON
 build_openbsd_TAR = gtar
 # openBSD touch doesn't understand -h
 build_openbsd_TOUCH = touch -m -t 200001011200
+
+# openbsd host on openbsd builder: override openbsd host preferences.
+openbsd_CC = clang
+openbsd_CXX = clang++

